src/command_modules/azure-cli-datalake/azure/cli/command_modules/datalake/analytics/custom.py

```diff
@@ -0,0 +1,229 @@
+                        tier=None):
+    adls_list = list()
+    adls_list.append(DataLakeStoreAccountInfo(default_datalake_store))
+    location = location or _get_resource_group_location(resource_group_name)
+    create_params = DataLakeAnalyticsAccount(location,
+                                             default_datalake_store,
+                                             adls_list)
+    if tags:
+        create_params.tags = tags
+
+    if max_degree_of_parallelism:
+        create_params.max_degree_of_parallelism = max_degree_of_parallelism
+
+    if max_job_count:
+        create_params.max_job_count = max_job_count
+
+    if query_store_retention:
+        create_params.query_store_retention = query_store_retention
+
+    if tier:
+        create_params.new_tier = tier
+
+    return client.create(resource_group_name, account_name, create_params)
+
+# pylint: disable=too-many-arguments
+def update_adla_account(client,
+                        account_name,
+                        resource_group_name,
+                        tags=None,
+                        max_degree_of_parallelism=None,
+                        max_job_count=None,
+                        query_store_retention=None,
+                        tier=None,
+                        firewall_state=None,
+                        allow_azure_ips=None):
+    update_params = DataLakeAnalyticsAccountUpdateParameters(
+        tags=tags,
+        max_degree_of_parallelism=max_degree_of_parallelism,
+        max_job_count=max_job_count,
+        query_store_retention=query_store_retention,
+        new_tier=tier,
+        firewall_state=firewall_state,
+        firewall_allow_azure_ips=allow_azure_ips)
+
+    return client.update(resource_group_name, account_name, update_params)
+
+# firewall customizations
+# pylint: disable=too-many-arguments
+def add_adla_firewall_rule(client,
+                           account_name,
+                           firewall_rule_name,
+                           start_ip_address,
+                           end_ip_address,
+                           resource_group_name):
+    create_params = FirewallRule(start_ip_address, end_ip_address)
+    return client.create_or_update(resource_group_name,
+                                   account_name,
+                                   firewall_rule_name,
+                                   create_params)
+
+# catalog customizations
+# pylint: disable=too-many-arguments
+def create_adla_catalog_credential(client,
+                                   account_name,
+                                   database_name,
+                                   credential_name,
+                                   credential_user_name,
+                                   uri,
+                                   credential_user_password=None):
+
+    if not credential_user_password:
+        try:
+            credential_user_password = prompt_pass('Credential Password:')
```

Comment:
Does this do the double prompt? I believe you need to set a kwarg for that. If not, I would recommend it.  On the command line you can see what you are putting in, but with prompt you don't see the value so doing a confirmation would be wise.